viewing_party/party.py

```diff
@@ -1,23 +1,179 @@
 # ------------- WAVE 1 --------------------
 
+from re import I
+
+
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+    if title == None or genre == None or rating == None:
+        return None
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+
+    user_data = {
+        "watched": [],
+        "watchlist": [],
+    }
+
+    add_to_watched(user_data, new_movie)
+    add_to_watchlist(user_data, new_movie)
+
+    return new_movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie):
+    for movie_dict in user_data["watchlist"]:
+        if movie_dict["title"] == movie:
+            user_data["watchlist"].remove(movie_dict)
+            user_data["watched"].append(movie_dict)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating_total = 0
+    num_watched = 0
+    if len(user_data["watched"]) == 0:
+        avg_rating = 0
+    else:
+        for movie_dict in user_data["watched"]:

```

Comment:
Notice that `user_data["watched"]` is used twice in this function, so a nice refactor might be to declare that in a variable.